internal/abci/abci.go

```diff
@@ -496,16 +479,30 @@ func (a *AbciApp) InitChain(ctx context.Context, req *abciTypes.RequestInitChain
 
 // ApplySnapshotChunk is on the state sync connection
 func (a *AbciApp) ApplySnapshotChunk(ctx context.Context, req *abciTypes.RequestApplySnapshotChunk) (*abciTypes.ResponseApplySnapshotChunk, error) {
-	refetchChunks, status, err := a.bootstrapper.ApplySnapshotChunk(req.Chunk, req.Index)
+	if a.statesyncer == nil {
+		return nil, errors.New("bootstrapper not set")
+	}
+
+	dbRestored, refetch, err := a.statesyncer.ApplySnapshotChunk(ctx, req.Chunk, req.Index)
 	if err != nil {
-		return &abciTypes.ResponseApplySnapshotChunk{Result: abciStatus(status), RefetchChunks: refetchChunks}, nil
+		var refetchChunks []uint32
```

Comment:
could you add a comment here explaining what is going on, specifically regarding refetch?